### PR TITLE
build: updating to use trunk based development

### PR DESCRIPTION
**Pull request by drduhe**
*Issue #, if available:*

*Description of changes:*
* Updating CONTRIBUTING.md to trunk based development guidance
* Updating workflows for trunk based development 
* Updating README.md to reflect changes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/pypi.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+on:
+  workflow_call:
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  deploy:
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
```
Comment (RanbirAulakh) [inline .github/workflows/pypi.yml:15]: Remove this if statement.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    uses: ./.github/workflows/docker-build.yml
+    secrets: inherit
+  pypi:
+    needs: [tox, docker-build]
+    uses: ./.github/workflows/pypi.yml
+    secrets: inherit
+  docker-publish:
+    needs: [tox, docker-build]
+    uses: ./.github/workflows/docker-publish.yml
+    secrets: inherit
```
Comment (RanbirAulakh) [inline .github/workflows/release.yml:21]: We don't have `docker-publish` workflow for this repo. I realized we don't even have a `dockerfile` in this repo. I recommend removing anything that's docker related. It looks like it got removed from this [commit](https://github.com/aws-solutions-library-samples/osml-model-runner-test/commit/d50c4e55eb33282298f2c2307a50b189c8b28c6e)

And update the readme to remove those as well unless we are planning to bring this back in.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  docs:
+    needs: [pypi, docker-publish]
```
Comment (RanbirAulakh) [inline .github/workflows/release.yml:23]: If we decided not to have `dockerfile`, would recommend removing `docker-publish`

File: CONTRIBUTING.md
Comment (RanbirAulakh) [inline CONTRIBUTING.md:1]: Also, remove the `images` folder as well.

File: .github/workflows/release.yml (hunk 2 above)
Comment (drduhe) [reply .github/workflows/release.yml]: Done

File: .github/workflows/release.yml (hunk 3 above)
Comment (drduhe) [reply .github/workflows/release.yml]: Done

File: CONTRIBUTING.md
Comment (drduhe) [reply CONTRIBUTING.md:1]: Done

**Review by RanbirAulakh — APPROVED** [review]

**Pull request merged**